src/vivarium_cluster_tools/psimulate/runner.py

```diff
@@ -66,37 +84,55 @@ def process_job_results(
     finally:
         batch_size = 500
         while unwritten_results:
-            written_results, unwritten_results = results.write_results_batch(
-                output_directory,
-                written_results,
+            (
+                existing_metadata,
+                unwritten_metadata,
+                existing_results,
+                unwritten_results,
+            ) = psim_results.write_results_batch(
+                output_paths,
+                existing_metadata,
+                existing_results,
+                unwritten_metadata,
                 unwritten_results,
                 batch_size=batch_size,
             )
             logger.info(f"Unwritten results: {len(unwritten_results)}")
             logger.info(f"Elapsed time: {(time() - start_time) / 60:.1f} minutes.")
-        return status
+
+    return status
 
 
-def load_existing_outputs(result_path: Path, restart: bool) -> pd.DataFrame:
+def load_existing_output_metadata(result_path: Path, restart: bool) -> pd.DataFrame:
     try:
-        existing_outputs = pd.read_hdf(result_path)
+        existing_output_metadata = pd.read_csv(result_path)
     except FileNotFoundError:
-        existing_outputs = pd.DataFrame()
+        existing_output_metadata = pd.DataFrame()
     assert (
-        existing_outputs.empty or restart
+        existing_output_metadata.empty or restart
     ), "How do you have existing outputs on an initial run?"
-    return existing_outputs
+    return existing_output_metadata
+
+
+def load_existing_results(result_path: Path, restart: bool) -> Dict[str, pd.DataFrame]:
+    filepaths = list(result_path.glob("*.parquet"))
+    results = {filepath.stem: pd.read_parquet(filepath) for filepath in filepaths}
+    if results and not restart:
+        raise RuntimeError(
+            f"This is an initial run but results aready exist at {result_path}"
+        )
+    return results
 
 
 def report_initial_status(
-    num_jobs_completed: int, existing_outputs: pd.DataFrame, keyspace: branches.Keyspace
+    num_jobs_completed: int, finished_sim_metadata: pd.DataFrame, total_num_jobs: int
 ) -> None:
     if num_jobs_completed:
         logger.info(
-            f"{num_jobs_completed} of {len(keyspace)} jobs completed in previous run."
+            f"{num_jobs_completed} of {total_num_jobs} jobs completed in previous run."
         )
-    if num_jobs_completed != len(existing_outputs):
-        extra_jobs_completed = num_jobs_completed - len(existing_outputs)
+    if num_jobs_completed != len(finished_sim_metadata):
+        extra_jobs_completed = num_jobs_completed - len(finished_sim_metadata)
         logger.warning(
```

Comment:
Out of scope, but are we sure this shouldn't be an error rather than a warning?

Reply:
Create a jira ticket to think about this